lib/gst/player/gstplayer.c

```diff
@@ -2641,6 +2658,46 @@ gst_player_seek_internal (gpointer user_data)
 }
 
 /**
+ * gst_player_set_rate:
+ * @player: #GstPlayer instance
+ * @rate: playback rate
+ *
+ * Playback at specified rate
+ */
+void
+gst_player_set_rate (GstPlayer *self, gdouble rate)
+{
+  g_return_if_fail (GST_IS_PLAYER (self));
+
+  self->rate = rate;
+  if (rate != 1.0)
+    self->trick_play = TRUE;
+  else
+    self->trick_play = FALSE;
+
+  self->seek_position = gst_player_get_position (self);
+
+  self->seek_source = g_idle_source_new ();
```

Comment:
Maybe check if a seek_source is already pending here. If so, the next call to gst_player_seek_internal() would already take the latest rate we set and we don't have to send yet another seek
I think we also need to use the mutex here to protect seek_source and seek_position.